File: app/src/main/java/com/android/bookingcucimotor/ListBookingAdapter.java
package com.android.bookingcucimotor;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;

public class ListBookingAdapter extends RecyclerView.Adapter<ListBookingAdapter.ListViewHolder> {

    private final ArrayList<Booking> listBooking;
    private OnItemClickCallback onItemClickCallback;

    public ListBookingAdapter(ArrayList<Booking> listBooking) {
        this.listBooking = listBooking;
    }

    public void setOnItemClickCallback(OnItemClickCallback onItemClickCallback) {
        this.onItemClickCallback = onItemClickCallback;
    }

    public static class ListViewHolder extends RecyclerView.ViewHolder {
        final ImageView imgPhoto;
        final TextView tvName;
        final TextView tvPrice;

        public ListViewHolder(@NonNull View itemView) {
            super(itemView);
            imgPhoto = itemView.findViewById(R.id.img_item_photo);
            tvName = itemView.findViewById(R.id.tv_item_name);
            tvPrice = itemView.findViewById(R.id.tv_item_price);
        }
    }

    @NonNull
    @Override
    public ListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list, parent, false);
        return new ListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ListViewHolder holder, int position) {
        Booking booking = listBooking.get(position);

        holder.imgPhoto.setImageResource(booking.getImg());
        holder.tvName.setText(booking.getName());
        holder.tvPrice.setText(booking.getPrice());

        holder.itemView.setOnClickListener(view -> {
            Intent intentDetail = new Intent(holder.itemView.getContext(), DetailActivity.class);
            intentDetail.putExtra("key_booking", listBooking.get(holder.getAdapterPosition()));
            holder.itemView.getContext().startActivity(intentDetail);
        });

        holder.itemView.setOnClickListener(view ->
                onItemClickCallback.onItemClicked(listBooking.get(holder.getAdapterPosition()))
        );
    }

    @Override
    public int getItemCount() {
        return listBooking.size();
    }

    public interface OnItemClickCallback {
        void onItemClicked(Booking data);
    }
}

File: app/src/main/java/com/android/bookingcucimotor/ListBookingAdapter.kt
package com.android.bookingcucimotor

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class ListBookingAdapter(private val listBooking: ArrayList<Booking>): RecyclerView.Adapter<ListBookingAdapter.ListViewHolder>() {

    private lateinit var onItemClickCallback: OnItemClickCallback

    fun setOnItemClickCallback(onItemClickCallback: OnItemClickCallback) {
        this.onItemClickCallback = onItemClickCallback
    }

    class ListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imgPhoto: ImageView = itemView.findViewById(R.id.img_item_photo)
        val tvName: TextView = itemView.findViewById(R.id.tv_item_name)
        val tvPrice: TextView = itemView.findViewById(R.id.tv_item_price)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_list, parent, false)
        return ListViewHolder(view)
    }

    override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
        val (imgResource, name, price) = listBooking[position]
        holder.imgPhoto.setImageResource(imgResource)
        holder.tvName.text = name
        holder.tvPrice.text = price

        holder.itemView.setOnClickListener{
            val intentDetail = Intent(holder.itemView.context, DetailActivity::class.java)
            intentDetail.putExtra("key_booking", listBooking[holder.adapterPosition])
            holder.itemView.context.startActivity(intentDetail)
        }
        holder.itemView.setOnClickListener{onItemClickCallback.onItemClicked(listBooking[holder.adapterPosition])}
    }

    override fun getItemCount(): Int = listBooking.size

    interface OnItemClickCallback {
        fun onItemClicked(data: Booking)
    }
}